### PR TITLE
compiler: Keep -qopenmp by default after icx 2023.2

### PR DESCRIPTION
**Pull request by georgebisbas**
Passes locally
After icx 2023.2.0, reductions should are passing

### CODE REVIEW

File: devito/arch/compiler.py [hunk 1]
```diff
@@ -761,10 +761,6 @@ def __init__(self, *args, **kwargs):
         platform = kwargs.pop('platform', configuration['platform'])
         language = kwargs.pop('language', configuration['language'])
 
-        if language == 'openmp':
```
Comment (mloubout) [inline devito/arch/compiler.py:764]: There is no guaranty an generic user will have icx >= 2023.2 the version number needs to be checked

Comment (georgebisbas) [reply devito/arch/compiler.py:764]: Right, thanks! I was thinking of our docker only! WIll fix

File: devito/arch/compiler.py [hunk 2]
```diff
@@ -761,7 +763,9 @@ def __init__(self, *args, **kwargs):
...
+        # Since OneAPI 2023.2.0 (clang17 underneath), we can use '-qopenmp'
+        # for all our tests
+        if self.version >= Version('17.0.0') and language == 'openmp':
```
Comment (FabioLuporini) [inline devito/arch/compiler.py:768]: isn't this doing the opposite of what the comment says?

also, if you are to change this (maybe not, maybe I misunderstood...), could you drop "for all our tests" which doesn't mean much? what you should have said is that earlier versions have an OpenMP bug. The bug is in their compilers, it doesn't have anything to do with our tests (I mean, not directly)

Comment (georgebisbas) [reply devito/arch/compiler.py]: damn sorry, you are right

File: devito/arch/compiler.py [hunk 3]
```diff
@@ -761,7 +763,8 @@ def __init__(self, *args, **kwargs):
         platform = kwargs.pop('platform', configuration['platform'])
         language = kwargs.pop('language', configuration['language'])
 
-        if language == 'openmp':
+        # Earlier to versions to OneAPI 2023.2.0 (clang17 underneath), have an OpenMP bug
```
Comment (FabioLuporini) [inline devito/arch/compiler.py:766]: typo, but fine, we can fix it in the next PR

**Review by FabioLuporini — APPROVED** [review]

**Review by mloubout — APPROVED** [review]

**Pull request merged**